spec/message.ebnf

```diff
@@ -21,6 +21,7 @@ Option ::= Name '=' (String | Nmtoken | Variable)
 /* Markup Tags */
 MarkupStart ::= Name Option*
 MarkupEnd ::= '/' Name
+MarkupEmpty ::= Name Option* '/'
```

Comment:
This isn't LL(1) because it conflicts with the `MarkupStart`. One way to fix this formally is something like the following (`MarkupStart` may need to be renamed):

```ebnf
MarkupStart ::= Name Option* '/'?
MarkupEnd ::= '/' Name
```

This should be OK because the EBNF only describes the _grammar_, not the data model. Assuming we end up with `ElementStart`, `ElementEnd`, and `ElementEmpty` in the data model, the parser can yield `ElementStart` when `MarkupStart` is missing the slash, and `ElementEmpty` when the slash is present.

Reply:
You're right. If keeping this syntax, it might make better to sense to have `Markup` and `MarkupEnd` in the syntax grammar, with `Markup` having a data model attribute like `isEmpty: boolean` determined by the presence (or not) of a trailing `/`.

The discussion in #269 is also relevant, as it may lead to different markup for Markup.

Reply:
I realize that the lack of symmetry between `Markup` and `MarkupEnd` may be bothersome. How about a single `Markup` production?

```ebnf
Markup ::= Name Option* '/'? | '/' Name
```

Reply:
While that certainly technically works, it does rather look like two separate constructs crammed into one.

Reply:
Right. EBNF requires a bit of formal accounting here and there to make it work, like the one I'm suggesting. Note that I only suggest to combine the _grammar productions_. The data model should have separate interfaces for all three element types.

Reply:
I updated the PR, merging MarkupStart and MarkupEmpty into one Markup element.